ruma-serde/src/boolean.rs

```diff
@@ -0,0 +1,81 @@
+//! De-/serialization functions for casting boolean values to int and vise versa.
+//! Delegates to `js_int::UInt` to ensure integer size is within bounds.
+
+use js_int::UInt;
+use serde::{
+    de::{self, Deserialize, Deserializer, Unexpected},
+    ser::{Error, Serialize, Serializer},
+};
+use std::convert::TryFrom;
+
+/// Serialize a boolean to UInt.
+pub fn serialize<S>(value: &bool, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    match UInt::try_from(match value {
+        false => 0,
+        true => 1,
+    }) {
```

Comment:
This can just be `UInt::from(value as u8)` or if that doesn't work `UInt::from(match value { false => 0u8, false => 1u8 })`. The conversion should not be fallible.